src/helics/apps/AsioBrokerServer.cpp

```diff
@@ -270,10 +270,12 @@ namespace apps {
     void AsioBrokerServer::mainLoop()
     {
         auto ioctx = AsioContextManager::getContextPointer();
+
 #ifdef HELICS_ENABLE_TCP_CORE
         if (tcp_enabled_) {
             tcpserver = loadTCPserver(ioctx->getBaseContext());
             tcpserver->setDataCall(
+                // NOLINTNEXTLINE
```

Comment:
What warning was caused by the next line?

Reply:
clang-tidy warning about the parameter being able to be passed by reference but it is a shared pointer that needs its lifetime extended so it shouldn't be passed by reference.